devito/tools/data_structures.py

```diff
@@ -113,7 +113,10 @@ def compare_to_first(v):
             if isinstance(first, np.ndarray) or isinstance(v, np.ndarray):
                 return (first == v).all()
             else:
-                return first == v
+                try:
+                    return v.issuperset({first})
```

Comment:
AKA 

```
first in v
```

?

Reply:
Well no (pushing a change wit hseparate test later). needs to be `v.issuperset(frozenset(first))` because `first` could be a set as well if there is multiple subdims in which case `first in v` would be false but `issuperset` would be true